### PR TITLE
gluon-mesh-batman-adv: remove gluon-mesh-batman-adv-14

### PR DESCRIPTION
**Pull request by NeoRaider**
In preparation for the update to batman-adv 2019.2 and its new UCI configuration, this removes batadv-legacy.

### CODE REVIEW

File: package/gluon-mesh-batman-adv/files/lib/netifd/proto/gluon_bat0.sh [hunk 1]
```diff
@@ -43,13 +43,7 @@ proto_gluon_bat0_setup() {
 	ip link set primary0 address "$primary0_mac" mtu 1532 up
 
 	local routing_algo="$(lua -e 'print(require("gluon.site").mesh.batman_adv.routing_algo())')"
-
-	if [ "$routing_algo" = 'BATMAN_IV_LEGACY' ]; then
-		modprobe batman-adv-legacy
-	else
-		modprobe batman-adv
```
Comment (blocktrron) [inline package/gluon-mesh-batman-adv/files/lib/netifd/proto/gluon_bat0.sh]: If we autoload the module now, shouldn't we get rid of the `rmmod batman-adv` in `proto_gluon_bat0_teardown`?

Comment (NeoRaider) [reply package/gluon-mesh-batman-adv/files/lib/netifd/proto/gluon_bat0.sh]: Fixed

**Comment by NeoRaider** [comment]
I have updated the FAQ as well.

**Comment by mweinelt** [comment]
Tested on my DIR860L-B1 with BATMAN_IV.

Node answers respondd queries, appears on the map. Neighbours/links look fine. Status page works.

I think it's fine, let's merge it.

**Pull request merged**